docs/src/autodiff.md

```diff
@@ -0,0 +1,5 @@
+# Gradient in AdvancedHMC.jl
+
+AdvancedHMC.jl supports automatic differentiation using [`LogDensityProblemsAD`](https://github.com/tpapp/LogDensityProblemsAD.jl) and user-specified gradients. While the default AD backend for AdvancedHMC.jl is ForwardDiff.jl, we can seamlessly change to other backend like Zygote.jl using various syntax like `Hamiltonian(metric, ℓπ, Zygote)`, `Hamiltonian(metric, ℓπ, Val(:Zygote))` or via ADTypes.jl `Hamiltonian(metric, ℓπ, AutoZygote())`.
```

Comment:
Might we also want to be opinionated in favouring the ADTypes approach, as the new standard way of specifying AD backends? I'm not suggesting leaving the other two out completely, but just putting ADTypes at the default solution and the other two as alternatives.

Reply:
Yeah, actually, I think the ADTypes approach is better since that would allow users to have more control in the AD backend. For example, we can specify `config` in the Mooncake backend, but the other two ways can not do such things.